### ISSUE TITLE
Setting Up a New Case Study

### ISSUE THREAD

**Issue by eascesharif** [rburghol/cbp_wsm#8]
To set up/run a new case study, ".riv" and ".land" must be created at: "./config/seglists". All the river segments must be listed in ".riv" file and all the connected land segments to them must be listed in ".land" file, even if some fall outside your watershed. Then, go to "./run/standard" and type this command: "csh run_all.csh p532cal_062211 PROJECTNAME", in which "PROJECTNAME" is the same as the ".land" and ".riv" file name.

**Comment by rburghol** [comment]
To be clear, if a river segment is included in the *.riv file, then by definition, it is in your watershed.  Then, all land segments that flow into any river segment that you have in the .riv file must be in the .land file.  

Also, you may look at the function:
`/run/make_seglists/basingen.csh`
All you need to do is specify the outlet river segments and it can assemble all contributing land and river segments for you.  If you are looking at a single watershed to model at a time, this is all you need to do.   Even if you need to run multiple watersheds, this function (if it works as advertised) will insure that you have the complete list of landsegs and riversegs.

```
usage: basingen.csh river_scenario basin

  BASIN can be a single unique ID, like 2720
     or a BASIN name, like P, YM, or EM1
     or the word ALL

```

**Comment by rburghol** [comment]
Oh -as for the ".calib" file - I don't know if that is needed or not?  I know that "basingen.csh" outputs it, so perhaps it is, but I had assumed it was only needed if you were recalibrating, but I may very well be wrong.

**Comment by eascesharif** [comment]
I don't quite understand "basingen". Should we repeat this 100 times if we have 100 river segments?

**Comment by rburghol** [comment]
Nope - just find the outlet segment ID, and run it once and it will assemble a list of all land and river segs contributing.  If you have multiple outlets you will have to set up multiple seglists, then merge those files if you wish to run them all simultaneously (though I do not encourage that).  For example - the following gets the Shenandoah River above its confluence with the Potomac River:
`./basingen.csh p532cal_060811 4380`
Run that and see - 4380 is the river segment ID for the outlet segment (see the GIS files of river segmentation)

**Comment by eascesharif** [comment]
For a case study that falls inside four major basins (J, P, R and Y), do I need to have four different lists?

**Comment by eascesharif** [comment]
And are "p532cal_060811" and "p532cal_062211" any different? The "'con" files are almost the same.

**Comment by rburghol** [comment]
The con files are roughly the same.  Per your question about thew 4 major basins, as I noted in previous comment, *I don't encourage* putting all basins in a single file, but it may run that way without problems (other than taking a longer time).

**Comment by rburghol** [comment]
Sorry - hit send to soon, I meant to respond to this:

> The "'con" files are almost the same.

I don't have first hand knowledge of the differences between scenarios.  The con files will describe the differences between the scenarios since they define all the parameters that make up those scenarios - so, the only way to know is to examine which parts are different, and then what is different about those parts.

**Comment by eascesharif** [comment]
My study area covers four basins: James, Potomac, Rappahannock and York, but only a small portion (four segments) of the York River is covered. However, after trying a few different things to run my study area, I feel like doing each basin (even subbasin) separately would be my best bet because all the upstream segments must be included in the river/land segments list. Do I need to pick up the results for each simulation (does the model overwrite the results) and compile them at a later point?

**Comment by rburghol** [comment]
I seldom use the formatted results from model runs, so I would auggest you try it and see. I can say the raw, unformatted results are segment specific so there should be no concern about one watershed overwriting another.  However, your different time periods WOULD get overwritten if he re-used the .con files and just changed land use files.

**Comment by eascesharif** [comment]
Although I was able to run the "basingen.csh" before, I get an error when using it (csh basingen.csh river_scenario 6120):

" PROBLEM FILE WRITTEN

Problem opening file:
../../../config/control/river/river_scenario.con
error =   2"

There was/is no such a file (river_scenario.con) in that directory and I'm not sure what this error is.

**Comment by eascesharif** [comment]
Since I can't use the "basingen", I started creating a small case manually. But I'm confused with the model. I tried running a single river segment: "YM2_6120_6430". It's located very upstream of the Pamunkey river and there's no segment upstream of it. Five land segments (A51033 A51137 A51177 F51033 F51177) are contributing to it. It's a calibration site segment too. But an error pops up when running the "scenario_postproc", since it looks for the downstream segment and don't find it. I thought all the upstream segments must be listed and downstream segments shouldn't matter. Or there might be something else that I'm missing...

**Comment by rburghol** [comment]
Basingen.csh is failing because you are not giving it a valid scenario name -- i.e. the error message is telling you that you are calling it incorrectly.  You are telling it to look for a scenario named "river_scenario", which means that you need to have 2 files to describe that scenario, The correct syntax is:
`./basingen.csh p532cal_060811 4380`

Where you must give it a name of an existing scenario (one defined in config/control/river/ and config/control/land).  I would use the example code that I gave until you get familiar with the operations, 

Have you successfully used basingen.csh per my above example?

**Comment by eascesharif** [comment]
thanks Rob. I made a silly mistake when running "Basingen". It works ok now :)

**Comment by rburghol** [comment]
Awesome!  What scenario did you choose to use?

**Comment by eascesharif** [comment]
I used "p532cal_062211"

**Comment by rburghol** [comment]
Cool. Keep me posted.

**Comment by eascesharif** [comment]
got the same error that I had before:

" PROBLEM FILE WRITTEN

  Problem opening
../../../output/del/tfs/aveann/p532cal_062211/YM3_6430_6620_p532
error code =   2"

I created the seglists for a single river segment: "YM2_6120_6430" via "Basingen". There's no segment upstream. The error happens at the very end module ("run_scenario_postproc.csh"). Seems like the model looks for the downstream river segment ("YM3_6430_6620"), but it's not there. Do we need to have the downstream segments too? I'm confused. Or is the error because of something else?

**Comment by rburghol** [comment]
Can you post the actual commands that you used so that I might verify?  Also, in the past, run_all.csh didn't complete the last command, but running the final command manually worked.

**Comment by eascesharif** [comment]
I ran each module separately:

1. csh run_lug.csh p532cal_062211 CLI_YM
2. csh run_rug.csh p532cal_062211 CLI_YM
3. csh run_land.csh p532cal_062211 CLI_YM
4. csh run_etm.csh p532cal_062211 CLI_YM
5. csh run_river.csh p532cal_062211 CLI_YM
6. csh run_scenario_postproc.csh p532cal_062211 CLI_YM

"CLI_YM" is the seglist name. got the error at #6.

**Comment by rburghol** [comment]
What command did you use to you run basingen? Also, I don't know if it matters but I don't use the "csh " method, instead just typing ./run_river.csh ...

**Comment by eascesharif** [comment]
I used this for "Basingen":

./basingen.csh p532cal_062211 6120

**Comment by eascesharif** [comment]
I tested running only the last module with "./" instead of "csh", but got the same error.

**Comment by rburghol** [comment]
Yeah the "csh " or "./" was just a shot in the dark.  As for my run - I got the same error as you when running "run_all.csh" - running the last 2 postproc scripts solo now - will report back.

**Comment by eascesharif** [comment]
it's weird to me. stmary case was very similar. it has only one river segment, with no upstream river segment. stmary runs ok, but this one does not!

**Comment by rburghol** [comment]
There could be something strange about this one -- I am running the shenandoah example now to see if anything different happens.  Just to be sure, did you reun the same scenario for stmary as this -- i.e. "p532cal_062211"

**Comment by eascesharif** [comment]
yes the same scenario because it was suggested in the "readme" file.

**Comment by eascesharif** [comment]
did you get an error when running the modules separately (my example YM_6120_6430)?

**Comment by rburghol** [comment]
> did you get an error when running the modules separately (my example YM_6120_6430)?

I did - now I am running Shenandoah to see if the same thing happens.

**Comment by eascesharif** [comment]
Looking into the "seglists" directory, there are some scenarios with "BASIN_order.riv" too. Do you know what it means and when it needs to be created? "Basingen" doesn't create it by default.

**Comment by eascesharif** [comment]
I also wonder how to get the daily, monthly and annual time series of loads and flow? Even for the stmary case, which runs fine, only aveann is generated. Does something need to be changed in the control file?

**Comment by rburghol** [comment]
All of the timeseries data are stored in wdm files - and you can use the following to convert to text:
'''/code/bin/quick_wdm_2_txt_hour_2_hour

**Comment by eascesharif** [comment]
any success on the Shenandoah run?

**Comment by rburghol** [comment]
Similar error - apparently looking for a segment that's NOT in the run:
```
 PROBLEM FILE WRITTEN

  Problem opening
../../../output/del/tfs/aveann/p532cal_062211/PS5_4370_4150_p532
error code =   2
```
Trying to isolate the problem now by running the last few lines of run_all.csh individually:

1. ./run_scenario_postproc.csh p532cal_062211 PS5_4380_4370
2. ./summarize_output_aveann.csh p532cal_062211 PS5_4380_4370  1991 2000
3.

**Comment by eascesharif** [comment]
where the output of "quick_wdm_2_txt_hour_2_hour" is generated? When I use it, I don't see a time series generated. The only thing's a WDU file, which only shows the directory path and has has no time series/values in it.

**Comment by eascesharif** [comment]
I think the reason why "stmary" case runs ok (the model doesn't ask for the downstream segment) is that the sole river segment drains to the bay.

**Comment by eascesharif** [comment]
does the model generate any sub-annual time series by default or should something (e.g., control file) be changed for it?

**Comment by rburghol** [comment]
> ... sub-annual time series by default
Not in the summary routines, but the model itself outputs many things at the hourly timestep into a wsm.  See the above comment, I included some info on the command: quick_wdm_2_txt_hour_2_hour SEG Yi Yf dsn

This is what we should learn how to use I think.  Also, I think your hypothesis may be correct.  It may be that running these CBP post-processing routines all expect that we want to do a summary of loads delivered to the bay (the whole reason for this model!).  Thus, they assume a full output.  So, I guess we need to learn how to harvest our own data for small runs, but I guess you DO want to know loads delivered to the bay?

**Comment by eascesharif** [comment]
I assumed that all the time series are generated by default, but looks like everything's generated at annual scale and the other folders (monthly and daily) are empty. Do you get the same?

And I'm not looking for delivered loads. All I need at this point is eos time series (loads, concentrations and flows).

**Comment by eascesharif** [comment]
I edited your comment on "quick_wdm_2_txt_hour_2_hour" command. I'm able to run it, but don't know where the outputs are located. I don't see any time series generated. The only thing's a WDU file, which only shows the directory path and has has no time series/values in it.

**Comment by rburghol** [comment]
The following directories contain the raw timeseries data wdm files for each segment:
tmp/wdm/land
tmp/wdm/river

There is a ton of stuff in there, including the eos that you are looking for, as well as the river segment inflow and outflow I think. For example, 
tmp/wdm/river/p532cal_062211/stream/PS5_5200_4380.wdm

Some information on what is in these files is here: documentation/informal/Model_Operations_Manual.doc

Let me know what you find -- I am going to look into exporting the data also.

**Comment by rburghol** [comment]
The DSN #s are a critical thing to know when outputting data from WDM files.  The uci files describe these, for example in the uci "", there are the following in the EXT TARGETS block (ext targets is outputs):

```
EXT TARGETS
<-Volume-> <-Grp> <-Member-><--Mult-->Tran <-Volume-> <Member> Tsys Tgap Amd ***
<Name>   #        <Name> # #<-factor->strg <Name>   # <Name> #  tem strg strg***
RCHRES   1 OFLOW  OVOL   3            SAME WDM4   111 WATR     ENGL      REPL
RCHRES   1 OFLOW  OHEAT  3            SAME WDM4   112 HEAT     ENGL      REPL
RCHRES   1 OFLOW  OXCF2  3 1          SAME WDM4   113 DOXY     ENGL      REPL
RCHRES   1 OFLOW  OSED   3 1          SAME WDM4   121 SAND     ENGL      REPL
RCHRES   1 OFLOW  OSED   3 2          SAME WDM4   122 SILT     ENGL      REPL
RCHRES   1 OFLOW  OSED   3 3          SAME WDM4   123 CLAY     ENGL      REPL
RCHRES   1 OFLOW  NUCF9  3 1          SAME WDM4   131 NO3D     ENGL      REPL
RCHRES   1 OFLOW  NUCF9  3 2          SAME WDM4   132 NH3D     ENGL      REPL
RCHRES   1 OFLOW  OSNH4  3 1          SAME WDM4   133 NH3A     ENGL      REPL
RCHRES   1 OFLOW  OSNH4  3 2          SAME WDM4   134 NH3I     ENGL      REPL
RCHRES   1 OFLOW  OSNH4  3 3          SAME WDM4   135 NH3C     ENGL      REPL
RCHRES   1 OFLOW  PKCF2  3 3          SAME WDM4   136 RORN     ENGL      REPL
RCHRES   1 OFLOW  NUCF9  3 4          SAME WDM4   141 PO4D     ENGL      REPL
RCHRES   1 OFLOW  OSPO4  3 1          SAME WDM4   142 PO4A     ENGL      REPL
RCHRES   1 OFLOW  OSPO4  3 2          SAME WDM4   143 PO4I     ENGL      REPL
RCHRES   1 OFLOW  OSPO4  3 3          SAME WDM4   144 PO4C     ENGL      REPL
RCHRES   1 OFLOW  PKCF2  3 4          SAME WDM4   145 RORP     ENGL      REPL
RCHRES   1 OFLOW  OXCF2  3 2          SAME WDM4   151 BODA     ENGL      REPL
RCHRES   1 OFLOW  PKCF2  3 5          SAME WDM4   152 TORC     ENGL      REPL
RCHRES   1 OFLOW  PKCF2  3 1          SAME WDM4   153 PHYT     ENGL      REPL
END EXT TARGETS
```

So to look at the outflow of water, I find DSN 111 - labelled here as "RCHRES 1 OFLOW OVOL", so I can use the quick_wdm_2_txt_hour_2_hour script like so:
```
/opt/model/p53/p532c-sova/code/bin/quick_wdm_2_txt_hour_2_hour
  program to write hourly ascii output from a wdm
 wdm name, start year, end year, dsn
PS5_5200_4380.wdm, 1990, 2000, 111
 hourly average =    154.04324018588326
 annual average =    1350427.0670550086

```
But as you can see, it only gives me an average value.  Not the detailed timeseries I was hoping for...

**Comment by eascesharif** [comment]
Bummer!

PS. I had pulled out the dsn# via WDMUtil software.

**Comment by rburghol** [comment]
Correction!  The script DOES output a CSV file of the timeseries data -- so when I do this:

```
cd /opt/model/p53/p532c-sova/tmp/wdm/river/p532cal_062211/stream
/opt/model/p53/p532c-sova/code/bin/quick_wdm_2_txt_hour_2_hour
PS5_5240_5200.wdm, 1984, 2005, 111
```
It creates a file called "PS5_5240_5200_0111.csv" with output in columns year, month, day, hour, value for the selected DSN.  I believe this is what we need :).

**Comment by eascesharif** [comment]
great! where's the csv located? Perhaps I don't run the command properly. Do you put a "./" before "quick_wdm_2_txt_hour_2_hour"?

**Comment by eascesharif** [comment]
I also don't see the time series of TSS, TP and TN in the WDMs when opening them with WDMUtil. dsn# 11 shows a non-zero time series for water constituent. The four others (ds# 111, 211, 311 and 411) are all zero and go until 2080.

**Comment by rburghol** [comment]
The csv appears wherever you run the file.  So, in my example above, I changed to the directory that has the file I wanted, then ran the command "quick_wdm_2_txt_hour_2_hour" by using it's absolute path (no "." beforehand), that is:
`/opt/model/p53/p532c-sova/code/bin/quick_wdm_2_txt_hour_2_hour`
Then it prompts me to enter the "wdm name, start year, end year, dsn" so I typed:
`PS5_5240_5200.wdm, 1984, 2005, 111`
And then it spit out a full summary, and the file that I requested (do an "ls -lhrt" and it will be the last file you see).

**Comment by rburghol** [comment]
I don't know if wdm util will give reliable data since it is windows, maybe a different version of HSPF.   Use the scripts from Linux and you will get what you want.  Also, make sure that you are using the correct path for the WDM files as there are many, some with the same names, and you want to make sure that you are getting the outputs, not inputs or transfer data.

As for TN, TSS and TP, the model doesn't output those as a lump sum, it outputs the different species that you need to add up to get your total.  That's what the run_summary* scripts do for you, but since you want real time data, you need to go with the csv version and add them up yourself -- or if you only need the annual output, the existing summary scripts do that just fine, you just will have to comment out the parts that do the annual loads delivered to the bay.

**Comment by eascesharif** [comment]
From the model documentation, I understand that all the time series (daily, monthly and annual) are generated in csv format in "output" directory if the model runs successfully. So, I think once we get the model running, those time series will be automatically generated in csv format. Am I right?

**Comment by rburghol** [comment]
Have you looked in there?

**Comment by eascesharif** [comment]
yes, but almost all the folders are empty. a few such as "aveann" under "river" and "annual" under "eos" are not.

**Comment by rburghol** [comment]
I suspect that's because our post-processing scripts quit prematurely?  But we should have some from stmary that completed, except that there still seems to be errors with that as well using run_all.csh -- though I don't know if that means the model didn't run, or that the post-processor didn't run correctly.

**Comment by eascesharif** [comment]
yes, there's nothing from stmary run in the "output" folder too, but there are some csv in the "sumout" directory.

what do you think we should do? I was hoping to get my baseline scenario results by the end of June... I'm a bit panic

**Comment by rburghol** [comment]
I think it's all their in front of you - the model data as it currently has run, can be harvested, with the CSV output I showed above.  If you think you want to rely on the CBP summary scripts then you need to figure out where the post-processing script is crashing and fix it.

**Comment by eascesharif** [comment]
I understand, but repeating this for more than 100 river segments that I have would require a very long time. And all of these must be repeated for 26 land use scenarios. So, I'm afraid that this isn't an efficient method for my case.

Will you be able to spend some time and help me with the "run_scenario_postproc.csh"?

**Comment by rburghol** [comment]
> repeating this for more than 100 river segments that I have would require a very long time
of course, but that's why we write batch scripts to handle things.
> Will you be able to spend some time and help me with the "run_scenario_postproc.csh"?
Most definitely, if you can take the lead I'll be glad to assist.

**Comment by eascesharif** [comment]
I'll try my best to figure it out.

Meanwhile, do the WDMs of river (./tmp/wdm/river/p532cal_062211/stream) show the real streamflow aggregated from all the contributing river and land segments?

**Comment by rburghol** [comment]
You can check the hspf docs to get an understanding of what each variable means, but the "EXT TARGETS" block generally contains the variables that are describing quantities at the reach outlet. The columns "GRP" and "Member Name" describe specific Hspf variable keywords.

**Comment by eascesharif** [comment]
I think "DFS" and "DEL" blocks in the postprocessing script ("run_scenario_postproc.csh") must be turned off when running a case study that doesn't drain to Bay. This tells the model not to look for downstream segments all the way to Bay. I was able to successfully run a single segment (that doesn't drain to Bay) via this.

**Comment by rburghol** [comment]
Sweet!  Does it then populate some of the more detailed data summaries -- like the directories output/river/daily/, and output/river/monthly/ or do we still need to do some more debugging of the summary scripts?

**Comment by eascesharif** [comment]
It does! For those, all the monthly and daily time series in the postprocessing shell script ("run_scenario_postproc.csh") must be turned on (1 instead of 0).

**Comment by rburghol** [comment]
Awesome work man!

**Comment by rburghol** [comment]
Can you provide a code example of how/where to turn the 0 to 1 for this?

**Comment by eascesharif** [comment]
Here's the one that I created:

[run_scenario_postproc_NoDEL.zip](https://github.com/rburghol/cbp_wsm/files/1131709/run_scenario_postproc_NoDEL.zip)

**Comment by rburghol** [comment]
Could you illustrate and explain your code here please -- no one wants to download a zip file and hunt for the changes that you made.

**Comment by eascesharif** [comment]
I set everything related to daily/monthly to 1. Everything in the "DFS" and "DEL" blocks was set to 0 too.

**Comment by rburghol** [comment]
What I was looking for was something like the following:
Changed run/standard/run_scenario_postproc.csh:
```
#  calculates delivery factors
  set DFS =          1
    set DFSmonthly = 0
    set DFSannual =  0
    set DFSaveann =  1

# calculates delivered load
  set DEL =          1
    set DELmonthly = 0
    set DELannual =  0
    set DELaveann =  1
```
To:
```
#  calculates delivery factors
  set DFS =          0
    set DFSmonthly = 0
    set DFSannual =  0
    set DFSaveann =  0

# calculates delivered load
  set DEL =          0
    set DELmonthly = 0
    set DELannual =  0
    set DELaveann =  0
```

**Comment by eascesharif** [comment]
thanks for the detailed description Rob.

**Comment by eascesharif** [comment]
could you run any case with these changes?

**Comment by rburghol** [comment]
I have a full Shenandoah River run going now - will report back when it finishes.

**Comment by eascesharif** [comment]
great! keep me posted. I'm running my entire case study too, which includes some parts of the Shenandoah River. It's pretty large case, so I expect a long computational time (more than a day).

**Comment by rburghol** [comment]
Had to make changes to 
From: 
```
  set EOF = 1
  set EOS = 1
  set DEL = 1
```
To;
 ```
 set EOF = 1
  set EOS = 1
  set DEL = 0
```

Which got me farther but still crashed at the end with an error that will need to be sorted:
 PROBLEM FILE WRITTEN

```
problem in transport factor program
 model run is earlier than allowed, change 'y1' in file
 pp/src/postproc/del/tfs/tfs.inc

```

**Comment by eascesharif** [comment]
In which script did you make this change?

Also, did you try to run all the modules or one by one?

**Comment by eascesharif** [comment]
Did you try running all the modules or one by one?

**Comment by eascesharif** [comment]
I was able to run my case study for all the modules except the very last (summarize_output_aveann.csh). I got the following error after running it for my case ("csh summarize_output_aveann.csh p532cal_062211 CLI_County_all 1991 2000"):

" PROBLEM FILE WRITTEN

Problem opening following file
../../../output/eof/aveann/p532cal_062211/A51049_to_JL1_7170_680"

The weird thing's that "JL1_7170_6800" exists in my river segment list, but seems like the model looks for the wrong segment ("JL1_7170_680" misses the last letter).

**Comment by rburghol** [comment]
That's a really interesting observation - I did not see the same thing in my test shenandoah run, but perhaps that has to do with the fact that I disabled DEL output?  Per your earlier question I was running each routine separately cause I was impatient.  

Can you post up a list of the changes that you made to the files?

I think it would soon be helpful to get you to pull a clone of this repository so that we can share git "diff" files on these to make sure that we have the same code base.

**Comment by rburghol** [comment]
OK - I was just experimenting with the date range, since the error that I got was "" - and I changed the start date to 1992 and got the same error that you got.  Not sure if that's related -- something I would note is that the 1992 test crashed almost immediately, while the 1991 iterated through several steps before crashing (I put more detailed output below so that you can see).

./summarize_output_aveann.csh p532cal_062211 PS5_4380_4370 1991 2000
```
 summarizing average annual output for scenario p532cal_062211 river basin PS5_4380_4370
PS1_4790_4830 PS2_6730_6660 PS2_5550_5560 PS1_4830_5080 PS0_6150_6160 PS2_5560_5100 PS3_5100_5080 PS2_6660_6490 PS4_5080_4380 PS3_6460_6230 PS3_5990_6161 PS2_6490_6420 PS0_6160_6161 PS2_6420_6360 PS3_6161_6280 PS3_6280_6230 PS4_6230_6360 PS4_6360_5840 PS4_5840_5240 PS5_5240_5200 PS5_5200_4380 PS5_4380_4370
 ../../../sumout/aveann/p532cal_062211/AllLoads_PS5_4380_4370_1991_2000_p532cal
 ../../../sumout/aveann/p532cal_062211/PS5_4380_4370_1991_2000_p532cal_062211_E
 ../../../sumout/aveann/p532cal_062211/PS5_4380_4370_1991_2000_p532cal_062211_E
 summarizing delivery factors for scenario p532cal_062211 river basin PS5_4380_4370


 PROBLEM FILE WRITTEN

problem in transport factor program
 model run is earlier than allowed, change 'y1' in file
 pp/src/postproc/del/tfs/tfs.inc
```
 ./summarize_output_aveann.csh p532cal_062211 PS5_4380_4370 1992 2000
```
 summarizing average annual output for scenario p532cal_062211 river basin PS5_4380_4370
PS1_4790_4830  ../../../output/eof/aveann/p532cal_062211/A51069_to_PS1_4790_4830_for_1992-2000.ave                                                  


 PROBLEM FILE WRITTEN

Problem opening following file
../../../output/eof/aveann/p532cal_062211/A51069_to_PS1_4790_483

```

**Comment by eascesharif** [comment]
I've disabled DEL too. I didn't make any further changes beyond what we discussed earlier. I edited your comment (three days ago) and that includes all the changes now. Let me know if you want me to provide more details.

**Comment by eascesharif** [comment]
I understand from the p532c documentation that only "etm", "river", "postproc" and "summarize" modules need to be run ("lug", "rug" and "land" modules can be skipped) if I only change the land use acreages. However, I wanted to double check and make sure that this is the correct way of doing it.

**Comment by eascesharif** [comment]
did you get a chance to dig into the summarize script? I looked into it but haven't found anything yet...

**Comment by rburghol** [comment]
I have not -counting on you here.  Have you gone through and printed out extra debugging information and determined where in the script it is failing, and what function call or sub-script is failing?

**Comment by eascesharif** [comment]
I don't know, but for me, it fails at the very beginning. I've made some changes to see what comes up this time. I'll keep you posted...

**Comment by rburghol** [comment]
I would make sure that it outputs the command that it is running at each step in the process -- insert your own echo statements to inform you of where it is (example below), and then when you get to an error, figure out which function is having the error and diagnose document it here.  IF I get a minute free I will try to look as well.
```

 # Insert the next line to advertise what command you are running
  echo "Running command: echo $scenario $basin $year1 $year2 $EOF $EOS $DEL | $tree/code/bin/sumout_aveann.exe "
  echo $scenario $basin $year1 $year2 $EOF $EOS $DEL | $tree/code/bin/sumout_aveann.exe

```

**Comment by rburghol** [comment]
I added that debug information for both of the sub-routines called in summarize_output_aveann.csh and it completes the first one OK (if I use 1991 as my start year), but failes when it gets to the 2nd subroutine (it only calls 2 functions in total).  The 2nd one is called "", and I don't even think it SHOULD call that, since we set the flag "DEL = 0", so I think we need to include logic in there to skip the 2nd one if DEL = 0 -- my guess is that since the bay program is ALWAYS focused on delivered loads, they never tested it with DEL = 0.

```
  echo "Running command: echo $scenario $basin $year1 $year2 $EOF $EOS $DEL | $tree/code/bin/sumout_aveann.exe "
  echo $scenario $basin $year1 $year2 $EOF $EOS $DEL | $tree/code/bin/sumout_aveann.exe
      if (-e problem) then
        echo ' '
        cat problem
        exit
      endif

  echo "Running command: echo $scenario $basin $year1 $year2 | $tree/code/bin/sum_delivery_factor.exe "
  echo $scenario $basin $year1 $year2 | $tree/code/bin/sum_delivery_factor.exe
      if (-e problem) then
        echo ' '
        cat problem
        exit
      endif


```

**Comment by eascesharif** [comment]
Sorry for the delay. I was testing something to see if that resolves this issue. The order of land segments was alphabetical (manually created). When I fixed that, the "summarize aveann" script took me further down and created the summary file, but failed later with this error:

" summarizing delivery factors for scenario p532cal_062211 river basin CLI_County_all


 PROBLEM FILE WRITTEN

problem in transport factor program
 model run is earlier than allowed, change 'y1' in file
 pp/src/postproc/del/tfs/tfs.inc"

This looks odd because I had turned off the delivery block.

**Comment by rburghol** [comment]
Regarding my comment above  (there is nothing in the current code to use the setting DEL = 0 in the routine sum_delivery_factor.exe) -- did you alter the code to have it skip the 2nd part where it calls "sum_delivery_factor.exe"?  If so please post up the changes so I can take a look -- but it looks like your code is not skipping it.

**Comment by eascesharif** [comment]
No Rob. I did not! I was so much focused on my run and didn't read your last comment.

But, I just got no error by simply commenting/removing the second loop.

FROM:

  echo $scenario $basin $year1 $year2 | $tree/code/bin/sum_delivery_factor.exe
      if (-e problem) then
        echo ' '
        cat problem
        exit
      endif

-----------------------------------------------------------------------------------------
TO:

#  echo $scenario $basin $year1 $year2 | $tree/code/bin/sum_delivery_factor.exe
#      if (-e problem) then
#        echo ' '
#        cat problem
#        exit
#      endif

**Comment by rburghol** [comment]
Well that makes sense -- but we should modify it to actually use the switch DEL = 0 since we will want to summarize loads at some time in the future.  But, that is very good news!

**Comment by eascesharif** [comment]
how about this?

FROM:

echo $scenario $basin $year1 $year2 | $tree/code/bin/sum_delivery_factor.exe
if (-e problem) then
echo ' '
cat problem
exit
endif

----------------------------------------------------------------------------------------
TO:

  if (DEL == 1) then  
  echo $scenario $basin $year1 $year2 | $tree/code/bin/sum_delivery_factor.exe
      if (-e problem) then
        echo ' '
        cat problem
        exit
      endif
  endif

**Comment by rburghol** [comment]
Looks promising - let me know if it works!

**Comment by eascesharif** [comment]
yes! worked for me. Did you test it too?

**Comment by rburghol** [comment]
Yup - works like a champ!

**Comment by rburghol** [comment]
Migrated that piece of code and committed to git repository on master branch.

**Comment by eascesharif** [comment]
since I want to get the time series (not only average) of the results at annual, monthly and daily intervals, I tried using the other "summarize" scripts (e.g., summarize_output_annual), but they need delivery methods, which I don't have in my case (it doesn't drain to Bay). Any thought to get them worked for this case?

There's also nothing to summarize daily loads...

**Comment by rburghol** [comment]
Unfortunately no.  Other than to run it all the way to the bay that is.
